code/datums/components/riding/riding.dm

```diff
@@ -101,6 +101,8 @@
 	if(isnull(.)) //you can set it to null to not change it.
 		. = AM.layer
 	AM.layer = .
+	for(var/mob/M AS in AM.buckled_mobs) //ensure proper layering of piggyback and carry, sometimes weird offsets get applied
+		M.layer = MOB_LAYER
```

Comment:
Explain please?

Reply:
![image](https://user-images.githubusercontent.com/25491240/130367588-781d0e1c-f6c9-428d-af21-c7e015bc6504.png)
 The exact same code was in human riding that you ported.
 
 Sometimes, randomly, the riding mob have a layer of 4.2 rather than 4 like it should.